src/main/java/org/folio/entlinks/utils/HeaderUtils.java

```diff
@@ -0,0 +1,20 @@
+package org.folio.entlinks.utils;
+
+import java.util.Arrays;
+import lombok.experimental.UtilityClass;
+import org.apache.kafka.common.header.Headers;
+
+@UtilityClass
+public class HeaderUtils {
+
+  public static String extractHeaderValue(String headerKey, Headers headers) {
```

Comment:
It would be better to return `Optional<String>` as empty in every case when you return an empty string.